main/src/library/Reader.flix

```diff
@@ -34,7 +34,7 @@ trait Reader[t] {
     ///
     /// Returns `Ok(k)` to signify that `k` bytes was successfully read and written to `b`.
     ///
-    /// Guarantees that `0 <= k < length(b)`.
+    /// Guarantees that `0 <= k <= length(b)`.
```

Comment:
I am confused. This suggested that we could read bytes past the length of the array.

Reply:
If we read every byte in the array, then the number of bytes read would be equal to the length of the array right? Since `k` is quantity and not index of last read byte. [Rust docs](https://doc.rust-lang.org/std/io/trait.Read.html#required-methods) use `<=` as well:

> If the return value of this method is [Ok(n)](https://doc.rust-lang.org/std/result/enum.Result.html#variant.Ok), then implementations must guarantee that 0 <= n <= buf.len()